backend/src/main/java/codesquard/app/api/wishitem/WishItemService.java

```diff
@@ -7,41 +7,49 @@
 import org.springframework.transaction.annotation.Transactional;
 
 import codesquard.app.api.errors.errorcode.ItemErrorCode;
-import codesquard.app.api.errors.errorcode.MemberErrorCode;
+import codesquard.app.api.errors.errorcode.WishErrorCode;
 import codesquard.app.api.errors.exception.RestApiException;
 import codesquard.app.api.response.ItemResponse;
 import codesquard.app.api.response.ItemResponses;
 import codesquard.app.domain.item.Item;
 import codesquard.app.domain.item.ItemRepository;
 import codesquard.app.domain.member.Member;
-import codesquard.app.domain.member.MemberRepository;
 import codesquard.app.domain.pagination.PaginationUtils;
 import codesquard.app.domain.wish.Wish;
 import codesquard.app.domain.wish.WishPaginationRepository;
 import codesquard.app.domain.wish.WishRepository;
+import codesquard.app.domain.wish.WishStatus;
 import lombok.RequiredArgsConstructor;
 
 @Service
 @RequiredArgsConstructor
 public class WishItemService {
 
-	private final MemberRepository memberRepository;
 	private final ItemRepository itemRepository;
 	private final WishRepository wishRepository;
 	private final WishPaginationRepository wishPaginationRepository;
 
 	@Transactional
-	public void register(Long itemId, Long memberId) {
+	public void changeWishStatus(Long itemId, Long memberId, WishStatus status) {
+		if (status == WishStatus.YES) {
+			register(itemId, memberId);
+			return;
+		}
+		cancel(itemId);
+	}
+
+	private void register(Long itemId, Long memberId) {
 		Item item = itemRepository.findById(itemId)
 			.orElseThrow(() -> new RestApiException(ItemErrorCode.ITEM_NOT_FOUND));
-		item.wishRegister();
-		Member member = memberRepository.findById(memberId)
-			.orElseThrow(() -> new RestApiException(MemberErrorCode.NOT_FOUND_MEMBER));
-		wishRepository.save(Wish.create(member, item, LocalDateTime.now()));
+		if (wishRepository.existsByMemberIdAndItemId(memberId, itemId)) {
+			throw new RestApiException(WishErrorCode.DUPLICATED_REQUEST);
+		} else {
+			item.wishRegister();
+			wishRepository.save(Wish.create(new Member(memberId), item, LocalDateTime.now()));
+		}
```

Comment:
```suggestion
		if (wishRepository.existsByMemberIdAndItemId(memberId, itemId)) {
			throw new RestApiException(WishErrorCode.DUPLICATED_REQUEST);
		}
		item.wishRegister();
		wishRepository.save(Wish.create(new Member(memberId), item, LocalDateTime.now()));
```

위와 같이 변경하면 불필요한 else를 사용할 필요가 없어보입니다.

Reply:
넵 수정하겠습니다~